apps/src/StudioApp.js

```diff
@@ -608,6 +608,61 @@ StudioApp.prototype.init = function (config) {
   }
 };
 
+StudioApp.prototype.hasValidContainedLevelResult_ = function () {
+  var results = this.getContainedLevelResults();
+  if (results.length !== 1) {
+    throw "Exactly one contained level result is currently required.";
+  }
+  var firstResult = results[0];
+  return typeof firstResult.result.response !== 'undefined' &&
+      firstResult.result.response !== '';
+};
+
+StudioApp.prototype.notifyInitialRenderComplete = function (config) {
+  if (config.hasContainedLevels && config.containedLevelOps) {
+    this.lockContainedLevelAnswers = config.containedLevelOps.lockAnswers;
+    this.getContainedLevelResults = config.containedLevelOps.getResults;
+
+    $('#containedLevel0').appendTo($('#containedLevelContainer'));
+
+    if (this.hasValidContainedLevelResult_()) {
+      // We already have an answer, don't allow it to be changed, but allow Run
+      // to be pressed so the code can be run again.
+      this.lockContainedLevelAnswers();
+    } else {
+      // No answers yet, disable Run button until there is an answer
+      $('#runButton').prop('disabled', true);
```

Comment:
Disabling `#runButton` after the initial render seems less-than-ideal, mostly because it's just one more pebble on the pile of things that manipulate the button on their own.  Is there any way this enable/disable logic can be integrated into one of the other ways the manipulate the button already?

Best case, can our initial render put the button in the appropriate state? (Maybe not viable - `#runButton` is initially rendered by React now, but is treated as 'protected' DOM and does not currently support later react updates, to avoid breaking everything that binds to it).

My second thought is that disabling the button is probably not sufficient to stop a clever student from running the program.  We may want to attempt a lower-level check for that (maybe you already did, and I haven't read far enough)!